### PR TITLE
refactor: Move processing of color from to_dict to a separate method.

### PR DESCRIPTION
**Pull request by benwoo1110**
## What type of pull request is this?

- [x] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition

## Changes

* Add a new `process_color` method.
* Embed color use export converter to `process_color` instead of overriding `to_dict`.

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

File: dis_snek/models/discord/color.py [hunk 1]
```diff
@@ -193,8 +195,22 @@ class FlatUIColors(Color, Enum):
     ASBESTOS = "#7F8C8D"
 
 
+def process_color(color: Union[Color, tuple, list, str, int]) -> int:
```
Comment (leestarb) [inline dis_snek/models/discord/color.py:198]: The dict type is missing

**Review by leestarb — APPROVED** [review]

File: dis_snek/models/discord/color.py (hunk 1 above)
Comment (leestarb) [inline dis_snek/models/discord/color.py]: The dict type is missing

**Review by leestarb — CHANGES_REQUESTED** [review]

File: dis_snek/models/discord/color.py [hunk 2]
```diff
@@ -193,8 +195,22 @@ class FlatUIColors(Color, Enum):
     ASBESTOS = "#7F8C8D"
 
 
+def process_color(color: Union[Color, dict, tuple, list, str, int]) -> int:
+    if not color:
```
Comment (artem30801) [inline dis_snek/models/discord/color.py:199]: is it for None/Missing? Then it should be reflected in argument type hint

Comment (benwoo1110) [reply dis_snek/models/discord/color.py:199]: aaa

**Pull request merged**